controller/konnect/reconciler_generic.go

```diff
@@ -491,28 +480,16 @@ func (r *KonnectEntityReconciler[T, TEnt]) Reconcile(
 		return ctrl.Result{}, nil
 	}
 
-	if res, err := ops.Update[T, TEnt](ctx, sdk, r.SyncPeriod, r.Client, r.MetricRecoder, ent); err != nil {
-		setServerURLAndOrgID(ent, serverURL, apiAuth.Status.OrganizationID)
-		if errUpd := r.Client.Status().Update(ctx, ent); errUpd != nil {
-			if k8serrors.IsConflict(errUpd) {
-				return ctrl.Result{Requeue: true}, nil
-			}
-			return ctrl.Result{}, fmt.Errorf("failed to update in cluster resource after Konnect update: %w %w", errUpd, err)
-		}
-
+	res, err = ops.Update[T, TEnt](ctx, sdk, r.SyncPeriod, r.Client, r.MetricRecoder, ent)
+	if res, errHandleStatus := handleStatusUpdate(ctx, ent, r.Client, serverURL, apiAuth.Status.OrganizationID, err); errHandleStatus != nil || !res.IsZero() {
+		return res, errHandleStatus
+	}
+	if err != nil {
 		return ctrl.Result{}, fmt.Errorf("failed to update object: %w", err)
 	} else if !res.IsZero() {
 		return res, nil
 	}
```

Comment:
It is hard to follow. Is there any reason why `ops.Update` can't include `handleStatusUpdate` or another way to encapsulate this? Because passing error as an argument and firstly checking result and error of `handleStatusUpdate` function and later going back in if-else chain to values returned by `ops.Update` has a lot of cognitive load

Reply:
ACK, how about e0942e1d14a46fccd1f2f56b9b72943f2cbaec06?